### ISSUE TITLE
"Frame is Too Long" error in Wireshark

### ISSUE THREAD

**Issue by flaviojs** [GNS3/dynamips#41]
### Description:

~~A packet with size >= 64KiB was dumped to the capture file but wireshark only suports up to 65535 bytes in a packet.~~

Sending and receiving packets is done in separate threads.
_libpcap_/_WinPcap_ isn't thread safe and two threads were dumping at the same time, resulting in the data of one of the frames being out of order. The next frame header contains garbage data, almost always leading to this error (the chance is over 99.99%).
### Analysis:

GNS3 uses the dynamips filter called "capture" to dump packets to a .cap file.
Which means I can probably fix it by splitting big packets into multiple chunks.

Different changes might be needed depending on what is triggering it:
- either it's sending a huge (tcp?) packet that is meant to be split by the network card during transmission
- or it's a bunch of packets sent together as one
- or another scenario I can't think of right now

---

All the limits I found:
- WinPcap NICs (Network Interface Cards) have a packet limit of 32KiB (BPF_MAXBUFSIZE in pcap/bfp.h)
- libpcap DLPIs (Data Link Provider Interfaces) have a packet limit of 32KiB or 64KiB
- libpcap filters have a packet limit of 50KiB
- the other cases aren't limited (analysis stuff ignored)

In short:
1. libpcap/WinPcap doesn't have a global packet size limit of 64KiB
2. the limits I found do not affect our particular case
3. the file format supports packets up to 4GiB in size (some places treat it as signed, so 2GiB is safer)

We are setting low snapshot values (max size of captured packets) and we don't respect it's value when dumping.
I'll change it to ~~be configurable,~~ default to the max value supported by wireshark and respect it while dumping.
NOTE: wireshark ignores the snapshot value

---

(continued in the next comment)
### Links:

http://forum.gns3.net/post31730.html
http://ask.wireshark.org/questions/23620/an-error-while-capturing-with-gns-3
